From fa678370aa5cd22df8260384da5bacc186b50550 Mon Sep 17 00:00:00 2001
From: Yinhang Liu <yinhang.liu@intel.com>
Date: Wed, 22 Feb 2023 10:41:40 +0800
Subject: [PATCH 100/205] msdkenc: Support image formats in low power mode for
 sink caps

There are some special image formats in low power mode, these
image formats should be supported in sink caps.

Part-of: <https://gitlab.freedesktop.org/gstreamer/gstreamer/-/merge_requests/4177>
---
 sys/msdk/gstmsdkcaps.c | 10 ++++++++--
 1 file changed, 8 insertions(+), 2 deletions(-)

diff --git a/sys/msdk/gstmsdkcaps.c b/sys/msdk/gstmsdkcaps.c
index 05c9fdba3..58a1ad793 100644
--- a/sys/msdk/gstmsdkcaps.c
+++ b/sys/msdk/gstmsdkcaps.c
@@ -648,8 +648,14 @@ _enc_is_format_supported (mfxSession * session,
   if (!_fill_mfxframeinfo (format, &in->mfx.FrameInfo))
     return FALSE;
 
-  if (!_enc_is_param_supported (session, in, out))
-    return FALSE;
+  in->mfx.LowPower = MFX_CODINGOPTION_UNKNOWN;
+  if (!_enc_is_param_supported (session, in, out)) {
+    in->mfx.LowPower = (out->mfx.LowPower == MFX_CODINGOPTION_ON) ?
+        MFX_CODINGOPTION_OFF : MFX_CODINGOPTION_ON;
+
+    if (!_enc_is_param_supported (session, in, out))
+      return FALSE;
+  }
 
   return TRUE;
 }
-- 
2.40.1

